torch/nn/functional.py

```diff
@@ -1777,8 +1777,24 @@ def multilabel_soft_margin_loss(input, target, weight=None, size_average=None,
     """
     if size_average is not None or reduce is not None:
         reduction = _Reduction.legacy_get_string(size_average, reduce)
-    input = torch.sigmoid(input)
-    return binary_cross_entropy(input, target, weight, None, None, reduction)
+
+    loss = - (target * logsigmoid(input) + (1 - target) * logsigmoid(-input))
+
+    if weight is not None:
+        loss = loss * weight
+
+    if reduction != 'none':
+        if reduction == 'sum':
+            return loss.sum()
+        else:  # elementwise_mean
+            return loss.mean()
```

Comment:
this is not element-wise mean, it's global mean.

Reply:
The [doc](https://github.com/pytorch/pytorch/blob/bfe2aa093e17c84db863672ad446de064e47dc7e/torch/nn/modules/loss.py#L889-L894) and [test](https://github.com/pytorch/pytorch/blob/9a9a7325c6f6b7f82636b6efdd742c9d1e63584e/test/test_nn.py#L6166-L6176) suggest this should be a global mean. So yeah, the args name is confusing...